File: src/test/java/color/GrayScaleTest.java
package color;

import cc.moecraft.logger.utils.ColorConverter;

/**
 * 此类由 Hykilpikonna 在 2018/12/16 创建!
 * Created by Hykilpikonna on 2018/12/16!
 * Github: https://github.com/hykilpikonna
 * QQ: admin@moecraft.cc -OR- 871674895
 *
 * @author Hykilpikonna
 */
public class GrayScaleTest
{
    public static void main(String[] args)
    {
        for (int r = 0; r < 256; r += 16)
        {
            System.out.print("\033[38;5;" + ColorConverter.rgbTo8Bit(r, r, r) + "m█");
        }
        
        System.out.println("");
        
        for (int r = 0; r < 256; r += 16)
        {
            System.out.print("\033[38;2;" + r + ";" + r + ";" + r + "m█");
        }
    }
}

File: src/test/java/color/GrayScaleTest.kt
package color

import org.hydev.logger.coloring.AnsiColorMode.TRUE_COLOR_24BIT
import org.hydev.logger.coloring.AnsiColorMode.XTERM_256_8BIT
import org.hydev.logger.foreground
import java.awt.Color

/**
 * 此类由 Hykilpikonna 在 2018/12/16 创建!
 * Created by Hykilpikonna on 2018/12/16!
 * Github: https://github.com/hykilpikonna
 * Meow!
 *
 * @author Hykilpikonna
 */
object GrayScaleTest
{
    @JvmStatic
    fun main(args: Array<String>)
    {
        run {
            for (r in 0..255 step 16)
            {
                print(Color(r, r, r).foreground(XTERM_256_8BIT) + "█")
            }
        }
        println("")
        for (r in 0..255 step 16)
        {
            print(Color(r, r, r).foreground(TRUE_COLOR_24BIT) + "█")
        }
    }
}
